frontend/src/app/main/ui/comments.cljs

```diff
@@ -1097,15 +1098,34 @@
                                                     groups))
          (group-bubbles zoom remaining visited (cons [current] groups)))))))
 
-(defn- calculate-zoom-scale-to-ungroup-bubbles
-  "Calculate the minimum zoom scale needed for a group of bubbles to avoid overlap among them"
-  [zoom threads]
+(defn- inside-vbox?
+  "Checks if a bubble or a bubble group is inside a viewbox"
+  [thread-group wl]
+  (let [vbox      (:vbox wl)
+        positions (mapv :position thread-group)
+        position  (gpt/center-points positions)
+        pos-x     (:x position)
+        pos-y     (:y position)
+        x1        (:x vbox)
+        y1        (:y vbox)
+        x2        (+ x1 (:width vbox))
+        y2        (+ y1 (:height vbox))]
+    (and (> x2 pos-x x1) (> y2 pos-y y1))))
+
+(defn- calculate-zoom-scale
+  "Calculates the zoom level needed to ungroup the largest number of bubbles while
+   keeping them all visible in the viewbox."
+  [position zoom threads wl]
   (let [num-threads         (count threads)
-        num-grouped-threads (count (group-bubbles zoom threads))
-        zoom-scale-step     1.75]
-    (if (= num-threads num-grouped-threads)
+        grouped-threads     (group-bubbles zoom threads)
+        num-grouped-threads (count grouped-threads)
+        zoom-scale-step     1.75
+        scaled-zoom         (* zoom zoom-scale-step)
+        zoomed-wl           (dwz/impl-update-zoom wl position scaled-zoom)]
+    (if (or (= num-threads num-grouped-threads)
+            (some #((comp not inside-vbox?) % zoomed-wl) grouped-threads))
```

Comment:
Consider define `(comp not inside-vbox?)` in a let binding and reuse, because with this code you are creating a new function on each iteration.

Also, this also can be simplified to `(complement inside-vbox?)`